src/v/cloud_storage/cache_service.cc

```diff
@@ -529,6 +529,8 @@ ss::future<cache::trim_result> cache::trim_fast(
     trim_result result;
 
     size_t candidate_i = 0;
+    const std::regex segment_expr{
+      R"#(.*\.log(\.\d+)?)#", std::regex_constants::optimize};
```

Comment:
nit: a comment on segment formats would be nice

Reply:
Added a comment, but I suspect there is a better way to document these expressions and strings which are spread around, perhaps the path name strings and associated patterns should live in a central specification namespace.